components/api/api-modules-javascript/src/main/resources/META-INF/dirigible/modules/src/http/client-async.ts

```diff
@@ -18,11 +18,21 @@ const HttpClientAsyncFacade = Java.type("org.eclipse.dirigible.components.api.ht
 const SpringBeanProvider = Java.type("org.eclipse.dirigible.components.spring.SpringBeanProvider");
 const httpClient = SpringBeanProvider.getBean(HttpClientAsyncFacade.class);
 
-export class HttpAsyncClient {
 
-	getAsync(url, config, options) {
-		const newUrl = buildUrl(url, options);
-		const callback = createHttpResponseCallback(
+interface Config {
+	success: Function;
+	error: Function;
+	cancel: Function;
+}
```

Comment:
Please set `readonly` for the properties that shouldn't be changed and also add `?` for the ones that are not required. For example:

```ts
interface Config {
	readonly success?: Function;
	readonly error?: Function;
	readonly cancel?: Function;
}
```